tenseal/tensors/ckksvector.cpp

```diff
@@ -204,9 +236,20 @@ CKKSVector& CKKSVector::mul_plain_inplace(vector<double> to_mul) {
     // prevent transparent ciphertext by adding a non-zero value
     if (to_mul.size() + 1 <= encoder->slot_count()) to_mul.push_back(1);
     encoder->encode(to_mul, this->init_scale, plaintext);
+
+    if (this->context->is_auto_mod_switch() &&
+        this->ciphertext.parms_id() != plaintext.parms_id()) {
+        set_to_same_mod(this->context, this->ciphertext, plaintext);
+    }
+
     this->context->evaluator->multiply_plain_inplace(this->ciphertext,
                                                      plaintext);
 
+    if (this->context->is_auto_rescale()) {
+        this->context->evaluator->rescale_to_next_inplace(this->ciphertext);
```

Comment:
same as above:
 - should we move the condition to a separate function?
 - should we check for last_parms_id?